app/models/reservation.rb

```diff
@@ -18,7 +18,8 @@ class Reservation < ActiveRecord::Base
   validate :matched_item_and_model
   validate :check_status
   validate :status_final_state
-  validate :not_in_past, :available, :check_banned, on: :create
+  validate :not_in_past, :available, :check_banned,
+           :check_consecutive, on: :create
```

Comment:
In retrospect, the fundamental problem we're trying to address here is *patrons* (or hypothetically checkout persons without override access) trying to bypass the rules by creating a set of overlapping or adjacent reservations. It's not a strict availability issue, so let's not worry about situations other than creation so the way it is now is fine. I was just focusing a bit too much on edge cases.